pandas/core/arrays/string_.py

```diff
@@ -205,9 +200,17 @@ def __setitem__(self, key, value):
 
         # validate new items
         if scalar_value:
-            if scalar_value is None:
-                value = np.nan
-            elif not (isinstance(value, str) or np.isnan(value)):
+            if (
+                value is libmissing.NA
+                or value is None
+                or (
+                    is_float_dtype(value)
+                    and not isinstance(value, str)
+                    and np.isnan(value)
```

Comment:
Can we use `pd.isna(..)` for all this instead? (we already checked that is a scalar) 
(that would also allow pd.NaT, though)